app/controllers/api/v1x0/mixins/rbac_mixin.rb

```diff
@@ -11,12 +11,16 @@ def read_access_check
           access_check('read')
         end
 
-        def access_check(verb)
+        def access_check(verb, id = params[:id])
```

Comment:
@lindgrenj6 we can break this method into multiple pieces
```ruby
def class_access_check(verb)
   return unless RBAC::Access.enabled?
  access_obj = RBAC::Access.new(controller_name.classify.constantize.table_name, verb).process
 raise Catalog::NotAuthorized, "#{verb.titleize}  access not authorized for #{controller_name.classify.constantize}" unless access_obj.accessible?
end

def instance_access_check(verb, id = nil)
.... current code ...
end
```

Reply:
@lindgrenj6 
RBAC uses something called permission which is the just the top level permission check
catalog:portfolios:read or catalog:portfolios:write

Then it uses the term resource to make it more granular to address each resource instance with  id.

so should we have 2 methods
permission_check(verb) 
resource_check with is the current access_check

Reply:
Updated with what I think you were saying. Let me know if that isn't what you were thinking.